simulator.go

```diff
@@ -266,14 +266,26 @@ func (h *simulatorAPIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
-			logger = logger.New("id", container.ID[:8])
+			cLogger := logger.New("id", container.ID[:8])
 
 			logfile := fmt.Sprintf("client-%s.log", container.ID[:8])
-			if _, err = runContainer(h.daemon, container.ID, logger, filepath.Join(h.logdir, logfile), false); err != nil {
+			waiter, err := runContainer(h.daemon, container.ID, cLogger, filepath.Join(h.logdir, logfile), false)
+			if err != nil {
 				logger.Error("failed to start client", "error", err)
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
+			go func() {
```

Comment:
Please don't start a new goroutine here. Instead, add a field to `simulatorAPIHandler` https://github.com/karalabe/hive/blob/master/simulator.go#L192 below `nodes` called `waiters`, that you can iterate over after closing the containers https://github.com/karalabe/hive/blob/master/simulator.go#L368. 

You'd also need to add the wait logic to the node `DELETE` handler https://github.com/karalabe/hive/blob/master/simulator.go#L340.

Having the extra field is cleaner that one more concurrent goroutine which may run at an arbitrary point in time.